qark/qarkMain.py

```diff
@@ -214,12 +214,19 @@ def process_manifest(manifest):
                 exit()
             common.logger.error(str(e) + "\r\nThat didnt work. Try providing an absolute path to the file\n")
 
+def hasmode(filename, mode):
+    st = os.stat(filename)
+    return (st.st_mode & mode)
+
+def setmode(filename, mode):
+    st = os.stat(filename)
+    os.chmod(filneame, st.st_mode | mode)
 
 def list_all_apk():
     result = []
     adb = common.getConfig('AndroidSDKPath') + "platform-tools/adb"
     st = os.stat(adb)
-    os.chmod(adb, st.st_mode | stat.S_IEXEC)
+    if not hasmode(adb, stat.S_IEXEC): setmode(adb, stat.S_IEXEC)
```

Comment:
This is a learning experience for me.  If you want me to refactor, let me know and I can start over.